File: forge-api/src/main/java/com/possible_triangle/flightlib/forge/api/ForgeFlightLib.java
package com.possible_triangle.flightlib.forge.api;

import com.possible_triangle.flightlib.api.IJetpack;
import net.minecraftforge.common.capabilities.Capability;
import net.minecraftforge.common.capabilities.CapabilityManager;
import net.minecraftforge.common.capabilities.CapabilityToken;

public class ForgeFlightLib {

    public static final Capability<IJetpack> JETPACK_CAPABILITY = CapabilityManager.get(new CapabilityToken<>() {
    });

}

File: forge-api/src/main/kotlin/com/possible_triangle/flightlib/forge/api/ForgeFlightLib.kt
package com.possible_triangle.flightlib.forge.api

import com.possible_triangle.flightlib.api.Constants
import com.possible_triangle.flightlib.api.IJetpack
import net.minecraft.resources.ResourceLocation
import net.neoforged.neoforge.capabilities.EntityCapability
import net.neoforged.neoforge.capabilities.ItemCapability

object ForgeFlightLib {

    @JvmField
    val ITEM_CAPABILITY: ItemCapability<IJetpack, Void?> = ItemCapability.createVoid(
        ResourceLocation.fromNamespaceAndPath(Constants.MOD_ID, "jetpack"),
        IJetpack::class.java,
    )

    @JvmField
    val ENTITY_CAPABILITY: EntityCapability<IJetpack, Void?> = EntityCapability.createVoid(
        ResourceLocation.fromNamespaceAndPath(Constants.MOD_ID, "jetpack"),
        IJetpack::class.java,
    )

}